sokoban/scripts/scene_menu.c

```diff
@@ -163,7 +163,18 @@ void menu_input_callback(InputKey key, InputType type, void* context)
         if (state == MenuState_Main)
             gameplayState->selectedCollection = 0;
         else if (state == MenuState_CollectionSelection)
+        {
+            LevelsCollection *coll = &database->collections[gameplayState->selectedCollection];
```

Comment:
Thanks! This is a very nice UX improvement. Two very small nitpicks:

1. Please rename `coll` to `collection` or `selectedCollection`.
2. Can you extract the loop into a method? `get_first_unplayed_level` or similar.

The final result would be:

```
LevelsCollection selectedCollection = ...
gameplayState->selectedLevel = get...
```

Reply:
Done!